File: tarock20-frontend/src/store/OpenGameDto.ts
export default class OpenGameDto {
    public id: string
    public players: string[]

    constructor(id: string, players: string[]) {
        this.id = id
        this.players = players
    }

    public isInGame(playerId: string): boolean {
        return this.players.includes(playerId)
    }
}

File: src/main/kotlin/eu/thelastdodo/tarock20/domain/OpenGameDto.kt
package eu.thelastdodo.tarock20.domain

import eu.thelastdodo.tarock20.entity.Game

data class OpenGameDto(
    val id: String,
    val players: List<String>,
) {
    constructor(game: Game): this(game.id, game.players.map { it.id })
}